src/jit/codegenxarch.cpp

```diff
@@ -1813,162 +1813,20 @@ CodeGen::genCodeForTreeNode(GenTreePtr treeNode)
                     assert(structDesc.eightByteCount == CLR_SYSTEMV_MAX_EIGHTBYTES_COUNT_TO_PASS_IN_REGISTERS);
 
                     regNumber retReg0 = REG_NA;
-                    emitAttr size0 = EA_UNKNOWN;
-                    unsigned offset0 = structDesc.eightByteOffsets[0];
+                    unsigned __int8 offset0 = 0;
```

Comment:
The varTypeIsStruct(tree) && tree->gtGetOp1()->OperGet() == GT_LCL_VAR change is also somewhat unrelated to the issues addressed by this PR. In the future, please open a separate issue for such cases, so we don't pile up unrelated things to a PR.